File: uva_10407.cpp
# include <iostream>
# include <vector>
using namespace std;

int gcd(int a,int b){
    while(b){
        int temp=a%b;
        a=b;
        b=temp;
    }
    return a;
}
int main(void){
    int data;
    vector<int> d;
    int first;
    while(true){
        d.clear();
        while(cin >> data){
            if (data){
                if (d.empty())
                    first=data;
                d.push_back(abs(data-first));
            }
            else 
                break;
        }
        int ans=0;
        if (d.empty())
            break;
        else
            for(int i = 0;i <d.size();i++)
                ans = gcd(d[i],ans);
        cout << ans << endl;
    }
    return 0;
}

File: uva_10407.py
def GCD(a, b):
    while b:
        temp=a%b
        if a>0 and temp<0:
            temp-=b
        a, b = b, temp
    return a

while True:
    d=[]
    data=input().split()
    if len(data)==1:
        break
    for item in data:
        d.append(abs(int(item)-int(data[0])))
    d=d[:-1]
    ans=0
    for item in d:
        ans=GCD(item, ans)
    print(ans)
